pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1223,6 +1226,55 @@ func (r *ReconcileHyperConverged) ensureIMSConfig(instance *hcov1alpha1.HyperCon
 	return r.client.Status().Update(req.ctx, instance)
 }
 
+func (r *ReconcileHyperConverged) ensureVMImport(instance *hcov1alpha1.HyperConverged, req *hcoRequest) error {
+	vmImport := newVMImportForCR(instance, req.Namespace)
+	if err := controllerutil.SetControllerReference(instance, vmImport, r.scheme); err != nil {
+		return err
+	}
+
+	key := client.ObjectKey{Namespace: "", Name: vmImport.GetName()}
+
+	found := &vmimportv1alpha1.VMImportConfig{}
+	err := r.client.Get(context.TODO(), key, found)
+	if err != nil && apierrors.IsNotFound(err) {
+		req.logger.Info("Creating vm import")
+		return r.client.Create(context.TODO(), vmImport)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	req.logger.Info("VM import exists", "vmImport.Namespace", found.Namespace, "vmImport.Name", found.Name)
+
+	// Add it to the list of RelatedObjects if found
+	objectRef, err := reference.GetReference(r.scheme, found)
+	if err != nil {
+		return err
+	}
+	objectreferencesv1.SetObjectReference(&instance.Status.RelatedObjects, *objectRef)
+
+	// Handle VMimport resource conditions
+	handleComponentConditions(r, req, "VMimport", found.Status.Conditions)
+
+	return r.client.Status().Update(context.TODO(), instance)
```

Comment:
Please use `req.ctx` instead of `context.TODO()`

Reply:
Done